src/main/java/com/codecool/virtualpub/ui/Display.java

```diff
@@ -1,4 +1,119 @@
 package com.codecool.virtualpub.ui;
 
+import com.codecool.virtualpub.data.*;
+
+import java.util.*;
+
 public class Display {
+
+    /**
+     * Displays a message to the console.
+     *
+     * @param message The message to be displayed.
+     */
+    public void displayMessage(String message) {
+        System.out.println(message);
+    }
+
+    /**
+     * Displays the bartender's script to the console.
+     *
+     * @param script The {@link BartenderScript} object containing the script to be displayed.
+     * @param index  The index of the script to be displayed from the list of bartender scripts.
+     */
+    public void displayScript(BartenderScript script, int index) {
+        try{
+            System.out.println("\n" + "Bartender: " + script.getScripts().get(index) + "\n");
+        } catch (Exception e) {
+            System.out.println("Script not found");
+        }
+    }
+
+    /**
+     * Displays the customer's script to the console.
+     *
+     * @param script The {@link CustomerScript} object containing the script to be displayed.
+     * @param index  The index of the sentence to be displayed from the list of customer sentences.
+     */
+    public void displayScript(CustomerScript script, int index) {
+        try{
+            System.out.println("\n" + "Customer: " + script.getSentences().get(index) + "\n");
+        } catch (Exception e) {
+            System.out.println("Script not found");
+        }
+    }
+
+    /**
+     * Displays the details of a drink (drink type and brand) to the console.
+     *
+     * @param drink The {@link Drink} object containing the drink details to be displayed.
+     */
+    public void displayDrink(Drink drink) {
+        System.out.println(drink.getDrinkType().getType() + ": " + drink.getBrand());
+    }
+
+    /**
+     * Displays the stock of drinks to the console. The drinks are grouped by {@link DrinkType}.
+     * Each drink type is followed by the available brands, their respective amounts, and the price per unit.
+     *
+     * @param drinks A list of {@link Drink} objects representing the stock of drinks.
+     */
+    public void displayStock(List<Drink> drinks) {
+        Map<DrinkType, StringBuilder> drinkTypeMap = new HashMap<>();
+
+        for (Drink drink : drinks) {
+            StringBuilder stringBuilder = drinkTypeMap.computeIfAbsent(drink.getDrinkType(), drinkType -> new StringBuilder("\n  " + drinkType + ":\n"));
```

Comment:
Holy cow I didn't know `computeIfAbsent` existed in Java! Awesome!

Also: very nice algorithm